Int32 current_number
Int32 previous_gesture
string current_action_block
string scopes
